### PR TITLE
feat: add PIV support and policy

### LINKED ISSUE #47: Implement PIV support
By using [ovh-yubico-piv-checker](https://github.com/ovh/yubico-piv-checker).
Parts of the support have already been open-sourced, namely:
- The `accountPIV` plugin, to set the per-account policy
- The `cron/osh-piv-grace-reaper.pl` script, to expire PIV grace periods
- The `accountInfo` plugin, to report the status of the account PIV policy
- a few internal helper functions

To get complete support we need to:
- add PIV support to `selfAddIngressKey`/`selfDelIngressKey`
- A global bastion-wide policy flag, if enforcing PIV for all accounts is desired

### PR DESCRIPTION
**Pull request by speed47**
Reviewing this PR commit by commit is advised.

Closes #47

### CODE REVIEW

File: bin/plugin/open/selfAddIngressKey [hunk 1]
```diff
@@ -38,18 +45,33 @@ if (ref $remainingOptions eq 'ARRAY' and @$remainingOptions) {
...
+if (($pivExplicit || $pivImplicitGlobal || $pivImplicitAccount) && (!OVH::Bastion::has_piv_helper())) {
+    if ($pivExplicit) {
+        osh_exit R('KO_PIV_NOT_AVAILABLE', msg => "This bastion doesn't have PIV capabilities due to missing prerequisites. Please retry without --piv.");
+    }
+    elsif ($pivImplicitGlobal) {
+        osh_exit R('KO_PIV_NOT_AVAILABLE',
+            msg => "This bastion is configured to only allow PIV keys but is missing some prerequisites to validate PIV keys. "
+              . "This is a configuration error, please contact your nearest sysadmin.");
+    }
+    elsif ($pivImplicitGlobal) {
```
Comment (Alkorin) [inline bin/plugin/open/selfAddIngressKey:61]: Twice the same "elsif"

Comment (Alkorin) [reply bin/plugin/open/selfAddIngressKey:61]:
```suggestion
    elsif ($pivImplicitAccount) {
```

**Review by Alkorin — CHANGES_REQUESTED** [review]
See https://github.com/ovh/the-bastion/pull/109/files#r550170765

**Review by Alkorin — APPROVED** [review]

**Comment by speed47** [comment]
I'll also implement the notion of account-scoped "never" policy that override the global setting, it's not that much work and will be useful.

**Pull request closed without merging**